### ISSUE TITLE
Servlet TCK secbasic#test7[_anno] depends on container specific behaviour

### ISSUE THREAD

**Issue by markt-asf** [jakartaee/platform-tck#44]
The two tests in question are:

- `src/com/sun/ts/tests/servlet/spec/security/secbasic/Client.java#test7`
- `src/com/sun/ts/tests/servlet/spec/security/secbasic/Client.java#test7_anno`

The tests are identical apart from the second tests uses annotations rather than web.xml for configuration.
The tests use a servlet mapped to `/ServletSecTest`. Note this is an exact mapping, not a path mapping.
There is a security constraint also mapped to `/ServletSecTest`
The client requests `/ServletSecTest/j_security_check` without any credentials
The tests expects a 401.
The reasoning is explained at https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/common/jspservletsec/SecBasicClient.java#L340

I do not think that the specification supports this behaviour.

I think the specification requires the following:
Since exact mapping is used, the request does not match the servlet mapping.
This means the request is passed to default servlet since it has a mapping of `/*`.
The servletPath will be the empty string and the pathInfo will be `/ServletSecTest/j_security_check`.
This doesn't match any security constraints so the default servlet will try to server this resource.
The resource does not exist so a 404 is returned.

The test could be 'fixed' by making the servlet mapping `/ServletSecTest/*` but that does not address the issue of the underlying premise of the test being incorrect. Changing the expected output (and the explanatory comment) to expect 404 is one possible solution. Simply removing/disabling the tests is another.

I'm happy to try and produce a PR for this once there is consensus on how it should be handled.

**Comment by janbartel** [comment]
I agree the test is flawed. At the very least the servlet and resource collection urls are wrong, and should have a trailing` '/*'`.  However, that said, I don't understand what is purported to be achieved with the request url containing a suffix of  '/j_security_check'.

**Comment by markt-asf** [comment]
On reflection, I think looking for 404 is the correct way to handle this.
The test is checking that the addition of "/j_security_check" doesn't trigger something in the authentication mechanism.

**Comment by arjantijms** [comment]
@markt-asf perhaps we could consider adding a test, for clarity, that a variant of 7 using "/blabla_check" instead of "/j_security_check" behaves exactly the same.
